### PR TITLE
DEP: integrate.quad_vec: deprecate `quadrature="trapz"`

### PR DESCRIPTION
**Pull request by j-bowhay**
#### Reference issue

towards #20486
#### What does this implement/fix?

Allows for cleaning up one of the final trapz references, which we have removed elsewhere as it is not an especially helpful acronym and can be interpreted as a slur.

Also tidies up some functions in the refguide check that no longer exist.
#### Additional information

Post to the development forum to follow once I do the other half of #20486

### CODE REVIEW

**Review by mdhaber — APPROVED** [review]
Looks good on my phone; I'll just test it locally before merging.

**Comment by mdhaber** [comment]
Documentation was already OK. Test looks good, and locally I see that we get the warning with `method='trapz'` and not with `method='trapezoid'`. Thanks @j-bowhay!

**Comment by mdhaber** [comment]
Ah, I see you intended to propose this on the forum before merging. I thought we had already decided that all these references would be removed, but it's up to you. LMK if you want this merged.

File: scipy/integrate/_quad_vec.py [hunk 1]
```diff
@@ -295,6 +296,12 @@ def quad_vec(f, a, b, epsabs=1e-200, epsrel=1e-8, norm='2', cache_size=100e6,
     except KeyError as e:
         raise ValueError(f"unknown quadrature {quadrature!r}") from e
 
+    if quadrature == "trapz":
+        msg = ("`quadrature='trapz'` is deprecated in favour of "
+               "`quadrature='trapezoid' and will raise a error from SciPy 1.16.0 "
```
Comment (h-vetinari) [inline scipy/integrate/_quad_vec.py:301]: typo
```suggestion
               "`quadrature='trapezoid' and will raise an error from SciPy 1.16.0 "
```

**Review by h-vetinari — APPROVED** [review]
LGTM aside from a typo

**Comment by j-bowhay** [comment]
Two approvals and no complaints from the mailing list so going to push the button on this.

**Pull request merged**